wily/commands/rank.py

```diff
@@ -0,0 +1,128 @@
+    :type metric: ''str''
+
+    :param revision_number: Version of git repository to revert to.
+    :type revision_number: ''int''
+
+    :return: Sorted table of all files in path, sorted in order of metric.
+    """
+    logger.debug("Running rank command")
+    logger.info(f"-----------Rank for {metric}------------")
+
+    data = []
+
+    operator, key = metric.split(".")
+    metric = resolve_metric(metric)
+    metric_meta = {
+        "key": key,
+        "operator": operator,
+        "title": metric.description,
+        "type": metric.type,
+        "wily_metric_type": metric.measure.name,  # AimHigh, AimLow, Informational
+        }
+
+    # vals = []
+    # Assumption is there is only one metric (e.g., therefore list of metrics commented out)
+    items = os.listdir(path)
+    for item in items:
+        full_item = os.path.join(path, item)
+        if os.path.isdir(full_item):
+            continue  #  sub-folders are ignored in first draft of rank
+        elif os.path.isfile(full_item):
+            state = State(config)
+            for archiver in state.archivers:
+                # Last revision in the list is the first item (ordered Newest to Oldest => 0 to -1 index.
+                rev = state.index[archiver].revisions[revision_number]
+                for meta in metric_meta:
+                    try:
+                        logger.debug(
+                            f"Fetching metric {meta['key']} for {meta['operator']} in {path}"
+                            )
+                        val = rev.get(config, archiver, meta["operator"], path, meta["key"])
+                        value = f"{val}"
+                    except KeyError as e:
+                        value = f"Not found {e}"
+                    # vals.append(k)
+                    # Assumption is there is only one metric (e.g., value versus val* as in report.py command
+                    data.append(
+                        (
+                            # wonder if revision specific information should be in here given that 
+                            # the focus is on the filename? Might not be a bad idea from an 
+                            # informational standpoint.
+                            item, #filename
+                            format_revision(rev.revision.key),
+                            rev.revision.author_name,
+                            format_date(rev.revision_date),
+                            # *vals,
+                            value,
+                        )
+                    )
+    # before moving towards the print tabular data - the values are sorted according to the metric type
+    # The "value" is assumed to be the fourth item in an individual data row. An alternative that may
+    # be more readable is the op.attrgetter.
+    if metric_meta["wily_metric_type"] == 'AimHigh':
+        # AimHigh is sorted lowest to highest
+        data.sort(key=op.itemgetter(4))
```

Comment:
nice, don't forget it's a namedtuple as well :-)

Reply:
Awesome...then it can be used with the attrgetter!

Reply:
is the namedtuple in reference to the data or the metric_meta variable - apologies if I am missing something clear/obvious?

The data variable seems to be a list   
`    logger.debug("Running rank command")
    logger.info(f"-----------Rank for {metric}------------")

    data = []

    operator, key = metric.split(".")`

whereas the metric_meta looks like a dictionary

    metric = resolve_metric(metric)
    metric_meta = {
        "key": key,
        "operator": operator,
        "title": metric.description,
        "type": metric.type,
        "wily_metric_type": metric.measure.name,  # AimHigh, AimLow, Informational
        }